CMSgov/easi-app
# [EASI-3030] Remove old grt_feedback table

# EASI-3030

## Changes and Description
### Backend:
- Replaces `grt_feedback` table with `governance_request_feedback` table.
  - All resolvers and store methods have been swapped out
  - V1 actions that previously saved `grt_feedback` now save a `governance_request_feedback` with some _assumed_ column values (since this table was built for V2, it doesn't perfectly match up with V1)
- DB Migration to migrate from `grt_feedback` to `governance_request_feedback`
  - Some assumptions were made about what column values make sense. Please review the migration with scrutiny for this reason!

### Frontend:
**EASI-3578**
- Removed `GetGRTFeedbackQuery`
  - Replaced all references in codebase with `GetGovernanceRequestFeedbackQuery`
- Updated `GetSystemIntakeQuery` to use `governanceRequestFeedbacks` field instead of deprecated `grtFeedbacks`
- Updates to handle nullable `author` field in `governanceRequestFeedbacks` type
  - If `author` is null, feedback renders "Governance Team" instead of "Author Name, Governance Team"

**EASI-3572**
- Viewing feedback from task list or forms now always renders new feedback page
  - Removed feature flag from v2 feedback page
  - Removed old feedback page
- Removed feature flag from new feedback admin tab 
- Removed feedback from business case views
  - Added info banner to admin business case view to show feedback has moved ([Figma](https://www.figma.com/file/ChzAP34A2DVvQUNQwD7lCt/IT-Governance-Next?type=design&node-id=4404%3A16165&mode=design&t=gnabFKMUYOzCKIdM-1))

## TODO


## How to test this change

To test frontend changes:
1. Complete intake form
2. As admin, complete action form to progress to draft business case step
    - Make sure feedback fields are filled in
3. Complete draft business case
4. Turn off `itGovV2Enabled` feature flag
5. Check that new feedback tab renders in admin view - /governance-review-team/:systemId/feedback
6. Check that info alert renders on admin business case view instead of feedback list - /governance-review-team/:systemId/business-case
7. Check that new feedback page renders for requester view - /governance-task-list/:systemId/feedback



## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
